src/algorithm/half_node.rs

```diff
@@ -126,7 +126,14 @@ mod test {
             .map(|(s, _)| s)
             .exactly_one()
             .unwrap();
-        let [&left,&right] = edge_classes.keys().filter(|(s,_)| *s == split).map(|(_,t)|t).collect::<Vec<_>>()[..] else {panic!("Split node should have two successors");};
+        let [&left, &right] = edge_classes
+            .keys()
+            .filter(|(s, _)| *s == split)
+            .map(|(_, t)| t)
+            .collect::<Vec<_>>()[..]
+        else {
+            panic!("Split node should have two successors");
```

Comment:
The frequency of this suggests we should have a helper method in `nest_cfgs::test` to do this. Or perhaps these would be better written using `collect_tuple` or some such. Probably not for this PR, but I could happily come up with something before/after/as an extra commit here ??